--- a/drivers/net/wireless/ath/ath11k/wmi.c
+++ b/drivers/net/wireless/ath/ath11k/wmi.c
@@ -7506,7 +7506,7 @@
 {
     struct wmi_pdev_bss_chan_info_event_tlv *tlv;
 
-    pdev_bss_chan_info->op_class = (u8)tlv->op_class;
+    pdev_bss_chan_info->op_mode = (u8)tlv->op_class;
     pdev_bss_chan_info->op_channel = (u8)tlv->op_channel;
     pdev_bss_chan_info->op_freq = (u16)tlv->op_freq;
     pdev_bss_chan_info->status = (u8)tlv->status;
